### PR TITLE
added support for Azure backend to Terraform deployment

### PR DESCRIPTION
**Pull request by eriksven**
Signed-off-by: Sven Erik Jeroschewski <svenerik.jeroschewski@bosch-si.com>

### CODE REVIEW

**Comment by eriksven** [comment]
@volsus83 what do you think?

File: deployment/terraform/README.md [hunk 1]
```diff
@@ -32,6 +32,10 @@ cp ./terraform.tfvars.template ./terraform.tfvars
 ```
 Edit ```./terraform.tfvars``` according your Azure subscription.   
 
+**configure Terraform backend**
+
+This deployment stores the state of in a storage in Azure. The storage is realized through a Terraform backend which needs to be configured in the main.tf. Nore information can be found under: https://www.terraform.io/docs/backends/types/azurerm.html.
```
Comment (selobosi) [inline deployment/terraform/README.md]: Only 2 minor typos:
- "the state of in storage" could be changed to "the state of Terraform in storage".
- "Nore information" is "More information"

File: deployment/terraform/variables.tf [hunk 2]
```diff
@@ -15,9 +15,9 @@ variable "SUBSCRIPTION_ID" {
 }
 
 variable "K8S_KUBE_CONFIG" {
-  description = "Path to Kube Config File"
+  description = "Path to Kube Config File (can be created through prepare environment)"
```
Comment (selobosi) [inline deployment/terraform/variables.tf]: "prepare environment" could be changed to "prepare_environment.sh" to make most clear what the name of the script is.

File: deployment/terraform/variables.tf [hunk 3]
```diff
@@ -15,9 +15,9 @@ variable "SUBSCRIPTION_ID" {
...
 }
 
 variable "K8S_HELM_HOME" {
-  description = "Path to Helm Home Directory"
+  description = "Path to Helm Home Directory (e.g. under $HOME/.helm) and can be created through prepare environment.sh"
```
Comment (selobosi) [inline deployment/terraform/variables.tf]: Thanks for adding this hint! "prepare environment.sh" could be changed to "prepare_environment.sh" to make most clear what the name of the script is.

**Pull request merged**